### PR TITLE
feat: download proton and eac runtime, switch to umu

### PR DESCRIPTION
**Pull request by imLinguin**
Currently it relies on `umu-run` script to be in PATH. I guess we should eventually download UMU launcher ourselves and fallback to that installed

### CODE REVIEW

File: maxima-lib/src/unix/wine.rs [hunk 1]
```diff
@@ -73,39 +108,67 @@ pub(crate) fn check_wine_validity() -> Result<bool> {
     Ok(version == release.unwrap().tag_name)
 }
 
-pub(crate) fn check_dxvk_validity() -> Result<bool> {
-    let version = versions()?.dxvk;
+fn get_lutris_runtimes() -> Result<Vec<LutrisRuntime>> {
+    let res = ureq::get("https://lutris.net/api/runtimes")
```
Comment (battledash) [inline maxima-lib/src/unix/wine.rs]: Use reqwest instead of ureq, it blocks the tokio runtime.

File: maxima-lib/src/unix/wine.rs [hunk 2]
```diff
@@ -73,39 +108,67 @@ pub(crate) fn check_wine_validity() -> Result<bool> {
...
+    let runtimes = get_lutris_runtimes()?;
+    let eac_runtime = runtimes.iter().find(|r| r.name == "eac_runtime");
+
+    Ok(eac_runtime.is_some_and(|r| r.created_at == version))
 }
 
-pub(crate) fn check_vkd3d_validity() -> Result<bool> {
-    let version = versions()?.vkd3d;
+pub(crate) fn install_eac() -> Result<()> {
+    log::info!("Downloading EasyAntiCheat runtime");
```
Comment (battledash) [inline maxima-lib/src/unix/wine.rs]: Unnecessary `log::`

File: maxima-lib/src/unix/wine.rs [hunk 3]
```diff
@@ -125,55 +188,28 @@ fn get_wine_release() -> Result<GithubRelease> {
...
     cwd: Option<PathBuf>,
     want_output: bool,
+    command_type: CommandType,
 ) -> Result<String> {
-    let path = maxima_dir()?.join(format!("wine/bin/{}", program));
-    //let path = PathBuf::from("umu-run");
+    let maxima_dir = maxima_dir()?;
+    let proton_path = maxima_dir.join("proton");
+    let proton_prefix_path = maxima_dir.join("prefix");
+    let eac_path = maxima_dir.join("eac_runtime");
```
Comment (battledash) [inline maxima-lib/src/unix/wine.rs]: Let's move `proton` and `eac_runtime` into `{maxima}/wine`, and have `{maxima}/wine/prefix` be the prefix dir.

**Review by battledash — CHANGES_REQUESTED** [review]

File: maxima-lib/src/unix/wine.rs (hunk 1 above)
Comment (imLinguin) [reply maxima-lib/src/unix/wine.rs]: We actually use ureq for github, that might need a refactor as well. Out of scope of this PR though..

Comment (battledash) [reply maxima-lib/src/unix/wine.rs]: Hmm, yeah. I'll see where else we use it and update them.

File: maxima-lib/src/unix/wine.rs [hunk 4]
```diff
@@ -187,17 +191,17 @@ fn get_wine_release() -> Result<GithubRelease> {
...
     want_output: bool,
     command_type: CommandType,
 ) -> Result<String> {
-    let maxima_dir = maxima_dir()?;
-    let proton_path = maxima_dir.join("proton");
-    let proton_prefix_path = maxima_dir.join("prefix");
-    let eac_path = maxima_dir.join("eac_runtime");
+    let wine_dir = maxima_dir()?.join("wine");
+    let proton_path = proton_dir()?;
+    let proton_prefix_path = wine_dir.join("prefix");
```
Comment (battledash) [inline maxima-lib/src/unix/wine.rs]: Use `wine_prefix_dir()`

File: maxima-lib/src/unix/wine.rs [hunk 5]
```diff
@@ -67,29 +68,38 @@ struct Versions {
     eac_runtime: String,
 }
 
+/// Returns internal prtoton pfx path
 pub fn wine_prefix_dir() -> Result<PathBuf> {
-    Ok(maxima_dir()?.join("prefix/pfx"))
+    Ok(maxima_dir()?.join("wine/prefix/pfx"))
```
Comment (battledash) [inline maxima-lib/src/unix/wine.rs]: This doesn't need `/pfx` anymore

File: maxima-lib/src/unix/wine.rs [hunk 6]
```diff
@@ -108,37 +118,31 @@ pub(crate) fn check_wine_validity() -> Result<bool> {
...
 
-pub(crate) fn install_eac() -> Result<()> {
-    log::info!("Downloading EasyAntiCheat runtime");
-    let runtimes = get_lutris_runtimes()?;
+pub(crate) async fn install_eac() -> Result<()> {
+    info!("Downloading EasyAntiCheat runtime");
+    let runtimes = get_lutris_runtimes().await?;
     let eac_runtime = runtimes.iter().find(|r| r.name == "eac_runtime").unwrap();
-    let path = maxima_dir()?.join("eac_runtime");
+    let path = maxima_dir()?.join("wine/eac_runtime");
```
Comment (battledash) [inline maxima-lib/src/unix/wine.rs]: `eac_dir()`

**Review by battledash — CHANGES_REQUESTED** [review]

File: maxima-lib/src/unix/wine.rs (hunk 5 above)
Comment (imLinguin) [reply maxima-lib/src/unix/wine.rs]: I initially kept it there for compatibility without UMU, but in the end we won't support a case like that. So it's okay to remove it

**Review by battledash — APPROVED** [review]
Thanks!

**Pull request merged**